src/components/Interpretations/InterpretationsProvider/InterpretationsManager.js

```diff
@@ -0,0 +1,307 @@
+        return () => {
+            this.interpretationObservers.get(id).delete(callback)
+        }
+    }
+
+    notifyInterpretationsListObserver() {
+        if (this.interpretationsListCallback) {
+            this.interpretationsListCallback(
+                groupInterpretationIdsByDate(
+                    Array.from(this.interpretations.values())
+                )
+            )
+        }
+    }
+
+    notifyInterpretationObservers(id) {
+        const callbacks = this.interpretationObservers.get(id)
+        if (callbacks) {
+            for (const callback of callbacks) {
+                callback(this.getInterpretation(id))
+            }
+        }
+    }
+
+    clearActiveInterpretation() {
+        this.activeInterpretationId = null
+    }
+
+    clearInterpretations() {
+        this.clearActiveInterpretation()
+        this.currentVisualizationId = null
+        this.currentType = null
+        this.interpretations.clear()
+    }
+
+    resetInterpretations(newInterpretations) {
+        this.interpretations.clear()
+        for (const interpretation of newInterpretations) {
+            this.interpretations.set(interpretation.id, interpretation)
+        }
+    }
+
+    async fetchInterpretationDetails(id) {
+        const result = await this.query({
+            interpretation: {
+                resource: 'interpretations',
+                id,
+                params: {
+                    fields: [
+                        'access[write,manage]',
+                        'comments[id,text,created,createdBy[id,displayName]]',
+                        'created',
+                        'createdBy[id,displayName]',
+                        'id',
+                        'likedBy',
+                        'likes',
+                        'text',
+                    ],
+                },
+            },
+        })
+        return result.interpretation
+    }
+
+    async fetchInterpretationsList() {
+        if (!this.currentType || !this.currentVisualizationId) {
+            throw new Error(
+                'Called fetchInterpretationsList before currentType or currentVisualizationId was set'
+            )
+        }
+        const result = await this.query({
+            interpretations: {
+                resource: 'interpretations',
+                params: {
+                    fields: [
+                        'access[write,manage]',
+                        'comments[id]',
+                        'created',
+                        'createdBy[id,displayName]',
+                        'id',
+                        'likedBy[id]',
+                        'likes',
+                        'text',
+                    ],
+                    filter: `${this.currentType}.id:eq:${this.currentVisualizationId}`,
+                    paging: false,
+                },
+            },
+        })
+        return result.interpretations.interpretations
+    }
+
+    async loadActiveInterpretation(id) {
```

Comment:
```suggestion
    async setActiveInterpretation(id) {
```
`load` is a bit ambiguous, I think `set` or `store` is more clear.